src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Registry/CosmosDbSearchParameterStatusDataStore.cs

```diff
@@ -39,13 +39,12 @@ public sealed class CosmosDbSearchParameterStatusDataStore : ISearchParameterSta
             _queryFactory = queryFactory;
         }
 
-        public async Task<IReadOnlyCollection<ResourceSearchParameterStatus>> GetSearchParameterStatuses()
+        public async Task<IReadOnlyCollection<ResourceSearchParameterStatus>> GetSearchParameterStatuses(CancellationToken cancellationToken = default)
         {
-            using var cancellationSource = new CancellationTokenSource(TimeSpan.FromMinutes(1));
             using IScoped<Container> clientScope = _containerScopeFactory.Invoke();
             DateTimeOffset startedCheck = Clock.UtcNow;
 
-            await _statusListSemaphore.WaitAsync(cancellationSource.Token);
+            await _statusListSemaphore.WaitAsync(cancellationToken);
```

Comment:
Having a default value for the CancellationToken here changes existing behavior (which is to have a CancellationToken that gets cancelled after 1 minute). Is this something we want to do for this method? If not, maybe we should add logic to check for the default value and handle appropriately?

Reply:
That is a fair point. In a few of the locations CancellationToken.None was being passed, so I took this as something that was put in place as a cancellation token wasn't being passed here. If there was an intent here to keep it short then we could look into additional logic. I don't have enough context to know if that is the case, so if @CaitlinV39 maybe knows, that would be helpful.

Reply:
@rotodd originally found this so I'll see if she has additional context.

Reply:
Changes in 038d041 look good to me. We will have two means of cancellation:

1. We will keep the existing behaviour to timeout after 1 minute (a thought: should we also have this for SQL?)
2. We will pass the cancellation token created at the top of the call stack into the database calls